components/emoji_picker/emoji_picker.tsx

```diff
@@ -0,0 +1,352 @@
+    useEffect(() => {
+        searchInputRef.current?.focus();
+    }, [visible]);
+
+    // clear out the active category on search input
+    useEffect(() => {
+        if (activeCategory !== getInitialActiveCategory()) {
+            setActiveCategory(getInitialActiveCategory());
+        }
+
+        resultsListRef?.current?.scrollToItem(0, 'start');
+    }, [filter]);
+
+    // scroll as little as possible on cursor navigation
+    useEffect(() => {
+        resultsListRef?.current?.scrollToItem(cursor.rowIndex, 'auto');
+    }, [cursor.rowIndex]);
+
+    const focusOnSearchInput = useCallback(() => {
+        searchInputRef.current?.focus();
+    }, []);
+
+    const getEmojiById = (emojiId: string) => {
+        if (!emojiId) {
+            return null;
+        }
+
+        const emoji = allEmojis[emojiId];
+        return emoji;
+    };
+
+    const handleCategoryClick = useCallback((categoryRowIndex: CategoryOrEmojiRow['index'], categoryIndex: number, categoryName: EmojiCategory, emojiId: string) => {
+        if (!categoryName || categoryName === activeCategory) {
+            return;
+        }
+
+        setActiveCategory(categoryName);
+        resultsListRef?.current?.scrollToItem(categoryRowIndex, 'start');
+
+        const cursorEmoji = getEmojiById(emojiId);
+        if (cursorEmoji) {
+            setCursor({
+                rowIndex: categoryRowIndex + 1, // +1 because next row is the emoji row
+                categoryIndex,
+                categoryName,
+                emojiIndex: 0, // first emoji of the category
+                emoji: cursorEmoji,
+            });
+        }
+    }, [activeCategory]);
+
+    const resetCursor = useCallback(() => {
+        setCursor({
+            rowIndex: -1,
+            categoryIndex: -1,
+            categoryName: '',
+            emojiIndex: -1,
+            emoji: undefined,
+        });
+    }, []);
+
+    const handleWithSearchKeyboardEmojiNavigation = (emojiIndex: EmojiCursor['emojiIndex']) => {
+        const numOfSearchRows = categoryOrEmojisRows.length - 1;
+        if (emojiIndex < 0 || emojiIndex > numOfSearchRows) {
+            return null;
+        }
+
+        const rowIndex = Math.ceil(emojiIndex / EMOJI_PER_ROW); // +1 because first row is search header
+
+        const emojiRow = categoryOrEmojisRows.find((row) => row.index === rowIndex + 1); // +1 because first row is search header
+        if (!emojiRow || emojiRow.type !== EMOJIS_ROW) {
+            return null;
+        }
+
+        const emojiIndexInsideRow = (emojiIndex % EMOJI_PER_ROW )+ 1;
+
+        const emoji = emojiRow.items?.[emojiIndexInsideRow] ?? null;
+        if (!emoji) {
+            return null;
+        }
+
+        setCursor({
+            rowIndex: rowIndex - 1,
+            categoryIndex: 0,
+            categoryName: emoji.categoryName,
+            emojiIndex,
+            emoji: emoji.item as Emoji,
+        });
+        return null;
```

Comment:
Looks like we can get rid of these returned values since they're always null

Reply:
✅  Updated as per request/discussion